include/opendht/network_engine.h

```diff
@@ -293,67 +241,179 @@ class NetworkEngine final
     /**
      * Sends values (with closest nodes) to a listenner.
      *
-     * @param sa  The address of the listenner.
-     * @param sslen  The length of the sockaddr structure.
-     * @param rid  The request id of the initial listen request.
-     * @param hash  The hash key of the value.
-     * @param want  Wether to send ipv4 and/or ipv6 nodes.
-     * @param ntoken  Listen security token.
-     * @param nodes  The ipv4 closest nodes.
-     * @param nodes6  The ipv6 closest nodes.
-     * @param values  The values to send.
+     * @param sa          The address of the listenner.
+     * @param sslen       The length of the sockaddr structure.
+     * @param socket_id  The tid to use to write to the request socket.
+     * @param hash        The hash key of the value.
+     * @param want        Wether to send ipv4 and/or ipv6 nodes.
+     * @param ntoken      Listen security token.
+     * @param nodes       The ipv4 closest nodes.
+     * @param nodes6      The ipv6 closest nodes.
+     * @param values      The values to send.
      */
-    void tellListener(std::shared_ptr<Node> n, uint16_t rid, const InfoHash& hash, want_t want, const Blob& ntoken,
+    void tellListener(std::shared_ptr<Node> n, uint32_t socket_id, const InfoHash& hash, want_t want, const Blob& ntoken,
```

Comment:
@aberaud. I was thinking about refactoring this part to a more generic code that would fit well with the new socket notion. I thought that, there could be an interface offered by the network engine equivalent to something like `NetworkEngine::write(Socket& socket)` but considering the fact that a unique function as such could not exist since it cannot know how to serialize the packets to write to the socket. I thought there could be some sort of model based on callbacks which would be defined at reception of the a request since such callbacks would be dependant to the request a node receives.

That being said, this patch as it is now is enough to be merged IMO. More enhancements could be added later.